src/connector/codec/src/decoder/utils.rs

```diff
@@ -33,6 +33,37 @@ pub fn scaled_bigint_to_rust_decimal(
     ))
 }
 
+/// Converts a Rust Decimal back to a `BigInt` with scale for Avro encoding
+pub fn rust_decimal_to_scaled_bigint(decimal: rust_decimal::Decimal) -> (BigInt, usize) {
+    let scale = decimal.scale() as usize;
+    let negative = decimal.is_sign_negative();
+
+    // Extract the parts
+    // let (lo, mid, hi, _, _) = decimal.into_parts();
+    let mut decimal_bytes = decimal.serialize();
+    // The serialize method returns the bytes in little-endian order so need to reverse it to fit the big-endian order
+    decimal_bytes.reverse();
+
+    // Concatenate bytes from hi, mid, and lo in big-endian order
+    let mut bytes = Vec::with_capacity(12);
+    bytes.extend_from_slice(&decimal_bytes[0..4]); // hi
+    bytes.extend_from_slice(&decimal_bytes[4..8]); // mid
+    bytes.extend_from_slice(&decimal_bytes[8..12]); // lo
+
+    // Trim leading zeros
+    let first_non_zero = bytes.iter().position(|&x| x != 0).unwrap_or(bytes.len());
+    bytes = bytes[first_non_zero..].to_vec();
+
+    // Create BigInt with correct sign
+    let bigint = if bytes.is_empty() {
+        BigInt::from(0)
+    } else {
+        BigInt::from_bytes_be(if negative { Sign::Minus } else { Sign::Plus }, &bytes)
```

Comment:
Consider convert to little-endian? Most of the machine now works on little-endian, and could be faster since BigInt provides `from_bytes_le`.

Reply:
I am thinking about ways to eliminate this step and directlyu build the bytes for avro decimal with big endian.